### DISCUSSION TITLE
Open modal after drop element

### DISCUSSION THREAD

**Discussion by throne1986** [GrapesJS/grapesjs#3758] [Q&A]
Hi @artf , how to show custom modal dialog when dropping elements on the editor?
like asset modal dialog appears when dropping the image block to the editor.

eg I want to have my own custom image block, so when the user drops this custom-image- block to canvas it should open a react-modal like this image below 

![image](https://user-images.githubusercontent.com/27904724/132480163-9f30048d-75d1-499f-8445-c607300fad5c.png)

tried like this below
         
     ........
     import ModalImages from '../components/ModalImages
     ........
          
        let modal = editor.Modal;
        
            editor.on("component:add", (model, argument) => {
              if (model.getName() == "image") {
                dispatch(setIsOpen(true));
                modal.setContent(ModalImages);
              }
            });
 
Unfortunately, when I drop the custom-image block nothing is happening, any idea what do I need to do to get this working?

**Comment by artf** [comment]
You can customize the `active` event of the component.
Here is a custom component with active method:
```js
editor.Components.addType('my-comp', {
  view: {
    onActive() {
      alert('Component activated');
      // Your logic here...
    }
  }
})
```
And this is how to configure your block to trigger activation on drop
```js
{
  label: 'Block label',
  content: { type: 'my-comp' },
  // ...
  activate: true,
}
```
